tests/cpp/test_abstract_tensor.cpp

```diff
@@ -712,4 +717,62 @@ TEST_F(AbstractTensorTest, UnzipBroadcasting) {
   EXPECT_EQ(ub[1], expect1);
 }
 
+TEST_F(AbstractTensorTest, TestApplyScheduling) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(1);
+  auto tv1 = makeSymbolicTensor(2);
+  auto tv2 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+  fusion.addInput(tv2);
+
+  // This fusion does not have a reference tensor
+  auto tv3 = broadcast(tv0, {false, true});
+  auto tv4 = mul(tv3, tv1);
+  auto tv5 = broadcast(tv0, {true, false});
+  auto tv6 = mul(tv5, tv2);
+
+  fusion.addOutput(tv4);
+  fusion.addOutput(tv6);
+
+  IdModel id_model(&fusion);
+  ValGraph& graph = id_model.idGraph(IdMappingMode::PERMISSIVE);
+  AbstractTensor abten;
+
+  abten.domain.reserve(3);
+  const auto addAbstractAxis = [&abten, &graph](IterDomain* id) {
+    ValGroup g = graph.toGroup(id);
+    abten.domain.emplace_back(ValGroupAndItsGraph{g, &graph});
+  };
+  addAbstractAxis(tv2->axis(0));
+  addAbstractAxis(tv1->axis(0));
+  addAbstractAxis(tv1->axis(1));
+
+  abten.merge(0);
+  abten.merge(0);
+  abten.split(0, 128);
+
+  for (TensorView* tv : {tv3, tv4, tv5, tv6}) {
+    applyAbstractSchedule(abten, tv, &graph);
+    tv->axis(-1)->parallelize(ParallelType::TIDx);
+    tv->axis(-2)->parallelize(ParallelType::BIDx);
```

Comment:
It might be nice to have a version of `scheduler_utils::parallelizeAllLike` that takes an `AbstractTensor` and vector of `ParallelType`s. That way, we wouldn't need to figure out which axes exist in `tv` here, which in general might mean replicating some of the logic in `applyAbstractSchedule`. Even better would be to be able to parallelize each axis as part of the `AbstractTensor` interface e.g.
```c++
abten.parallelize(-1, ParallelType::Vectorize);
// ...
parallelizeAllLike(/*reference=*/abten);
```